File: nodejs/patterns/src/creational/singleton.js
class Configuration {
  static instance;

  constructor() {
    if (Configuration.instance) {
      return Configuration.instance;
    }
    this.settings = new Map();
    Configuration.instance = this;
  }

  set(key, value) {
    this.settings.set(key, value);
  }

  get(key) {
    return this.settings.get(key);
  }
}

function demo() {
  const first = new Configuration();
  const second = new Configuration();
  first.set("mode", "prototype");
  return first === second && second.get("mode") === "prototype";
}

module.exports = { Configuration, demo };

File: python/patterns/src/prototype_patterns/creational/singleton.py
"""Singleton pattern example."""

from __future__ import annotations


class Configuration:
    """Ensures only one configuration instance exists."""

    _instance: Configuration | None = None

    def __new__(cls) -> Configuration:
        if cls._instance is None:
            cls._instance = super().__new__(cls)
            cls._instance._settings = {}
        return cls._instance

    def set(self, key: str, value: str) -> None:
        self._settings[key] = value

    def get(self, key: str) -> str | None:
        return self._settings.get(key)


def demo() -> bool:
    first = Configuration()
    second = Configuration()
    first.set("mode", "prototype")
    return first is second and second.get("mode") == "prototype"
